# Mice

Mice were injected with a single-stranded construct: CMV-eGFP-BGHpA_stuffer, packaged in either AAV2 or AAV9. 3 Groups of 9 mice each. In every group there are 4 mice each injected with 5e11 AAV2 and AAV9, as well as 1 mouse injected with PBS. Group 1 was sacrificed after 2 weeks, Group 2 after 6 weeks, and Group 3 12 weeks post injection.

# Sample processing

In this dataset I have included 2 runs of Cut and Tag library preparation, which are in total 14 (CT14-CT27). If everything worked out I will have sequenced DNA from 3 mice each from every timepoint and AAV injection.

For these samples a new Active Motif kit was used.

## Samples 

Used mice from all different groups. Samples from multiple mice each. E.g. 2-6-1 = AAV2, 6 weeks p.i., mouse nr 1

Samples CT14 to CT19 were prepared in the first batch on 5.9.24

Samples CT20 to CT27 were prepared in the second batch on 9.9.24


## Library preparation

Followed the ActiveMotif Cut and Tag IT Tissue Kit to extract chromatin and perform the Cut and Tag method.

Quantified the nuclei suspension on a Luna Cell Counter and used ~250000 Nuclei for Library prep.

Used the Rabbit@H3K27ac primary ab and GuineaPig@Rabbit sevondary ab, all from the kit.

The used index primers (from the kit) are noted in the table below, as well as Bioanalyzer results of the finished libraries.

Pooled 100 fmol of every sample in the final library that was handed to David Ibberson. CT27 was concentrated too low, therefore only 35 fmol of that.


| CT   | mouse  | nuc/mL | i7 | i5 | peak (bp) | ng/ÂµL | nM   |
|------|--------|--------|----|----|-----------|-------|------|
| CT14 | 2-12-1 | 1.8e6  | 1  | 1  | 216       | 2.66  | 18.7 |
| CT15 | 2-12-2 | 2.4e6  | 2  | 2  | 222       | 3.38  | 23.0 |
| CT16 | 2-12-3 | 3.7e6  | 3  | 3  | 337       | 6.43  | 28.9 |
| CT17 | 9-12-1 | 2.1e6  | 4  | 4  | 321       | 3.6   | 17.0 |
| CT18 | 9-12-2 | 2.1e6  | 1  | 4  | 211       | 6.66  | 47.9 |
| CT19 | 9-12-3 | 2.2e6  | 2  | 3  | 329       | 5.15  | 23.7 |
| CT20 | 2-2-1  | 2.9e6  | 1  | 2  | 330       | 3.83  | 17.6 |
| CT21 | 2-2-2  | 2.4e6  | 2  | 1  | 354       | 5.76  | 24.6 |
| CT22 | 2-2-3  | 3.8e6  | 3  | 4  | 317       | 1.78  | 8.5  |
| CT23 | 2-6-3  | 3.5e6  | 4  | 3  | 355       | 3.44  | 14.7 |
| CT24 | 9-2-1  | 2.1e6  | 4  | 1  | 324       | 2.38  | 11.1 |
| CT25 | 9-2-2  | 3.0e6  | 3  | 2  | 331       | 1.54  | 7.0  |
| CT26 | 9-2-3  | 5.0e6  | 1  | 3  | 352       | 4.09  | 17.6 |
| CT27 | 9-6-3  | 8.0e6  | 4  | 2  | 527       | 1.23  | 3.5  |


## Sequencing

Sequencing was performed by the Deep Sequencing Core and David Ibberson. Was run on 21.9.2024 with a set of the P2 SBS reagents on the NextSeq2000.

Demultiplexing performed by David.


# Sample renaming

```{bash}
declare -A barcode_dict=(
  ["CT20"]="K27ac_2-2-1_liver"
  ["CT21"]="K27ac_2-2-2_liver"
  ["CT22"]="K27ac_2-2-3_liver"
  ["CT24"]="K27ac_9-2-1_liver"
  ["CT25"]="K27ac_9-2-2_liver"
  ["CT26"]="K27ac_9-2-3_liver"
  ["CT23"]="K27ac_2-6-3_liver"
  ["CT27"]="K27ac_9-6-3_liver"
  ["CT14"]="K27ac_2-12-1_liver"
  ["CT15"]="K27ac_2-12-2_liver"
  ["CT16"]="K27ac_2-12-3_liver"
  ["CT17"]="K27ac_9-12-1_liver"
  ["CT18"]="K27ac_9-12-2_liver"
  ["CT19"]="K27ac_9-12-3_liver"
)

for R in R1 R2
do
  for key in ${!barcode_dict[@]}
  do
    infile=~/sds-hd/sd21j005/conradin/raw_data/illumina/nextseq/20240923_cutandtag_03/${key}_S*_${R}_001.fastq.gz
    outfile=~/sds-hd/sd21j005/conradin/projects/seq_runs/cutandtag_03/data/raw_data/${barcode_dict[$key]}_${R}.fastq.gz
    cp $infile $outfile
  done
done
```


In total there are 532 million reads. Minimum 18 million reads and maximum 53 million. Comparing to the bioanalyzer profiles I can not see a big correlation with the yield.

| file                           | format | type | num_seqs   | sum_len       | min_len | avg_len | max_len |
|--------------------------------|--------|------|------------|---------------|---------|---------|---------|
| K27ac_2-12-1_liver_R1.fastq.gz | FASTQ  | DNA  | 18,002,016 | 1,098,122,976 | 61      | 61      | 61      |
| K27ac_2-12-1_liver_R2.fastq.gz | FASTQ  | DNA  | 18,002,016 | 1,098,122,976 | 61      | 61      | 61      |
| K27ac_2-12-2_liver_R1.fastq.gz | FASTQ  | DNA  | 23,143,826 | 1,411,773,386 | 61      | 61      | 61      |
| K27ac_2-12-2_liver_R2.fastq.gz | FASTQ  | DNA  | 23,143,826 | 1,411,773,386 | 61      | 61      | 61      |
| K27ac_2-12-3_liver_R1.fastq.gz | FASTQ  | DNA  | 45,924,303 | 2,801,382,483 | 61      | 61      | 61      |
| K27ac_2-12-3_liver_R2.fastq.gz | FASTQ  | DNA  | 45,924,303 | 2,801,382,483 | 61      | 61      | 61      |
| K27ac_2-2-1_liver_R1.fastq.gz  | FASTQ  | DNA  | 53,081,728 | 3,237,985,408 | 61      | 61      | 61      |
| K27ac_2-2-1_liver_R2.fastq.gz  | FASTQ  | DNA  | 53,081,728 | 3,237,985,408 | 61      | 61      | 61      |
| K27ac_2-2-2_liver_R1.fastq.gz  | FASTQ  | DNA  | 36,575,553 | 2,231,108,733 | 61      | 61      | 61      |
| K27ac_2-2-2_liver_R2.fastq.gz  | FASTQ  | DNA  | 36,575,553 | 2,231,108,733 | 61      | 61      | 61      |
| K27ac_2-2-3_liver_R1.fastq.gz  | FASTQ  | DNA  | 45,509,490 | 2,776,078,890 | 61      | 61      | 61      |
| K27ac_2-2-3_liver_R2.fastq.gz  | FASTQ  | DNA  | 45,509,490 | 2,776,078,890 | 61      | 61      | 61      |
| K27ac_2-6-3_liver_R1.fastq.gz  | FASTQ  | DNA  | 47,553,412 | 2,900,758,132 | 61      | 61      | 61      |
| K27ac_2-6-3_liver_R2.fastq.gz  | FASTQ  | DNA  | 47,553,412 | 2,900,758,132 | 61      | 61      | 61      |
| K27ac_9-12-1_liver_R1.fastq.gz | FASTQ  | DNA  | 48,440,536 | 2,954,872,696 | 61      | 61      | 61      |
| K27ac_9-12-1_liver_R2.fastq.gz | FASTQ  | DNA  | 48,440,536 | 2,954,872,696 | 61      | 61      | 61      |
| K27ac_9-12-2_liver_R1.fastq.gz | FASTQ  | DNA  | 21,594,265 | 1,317,250,165 | 61      | 61      | 61      |
| K27ac_9-12-2_liver_R2.fastq.gz | FASTQ  | DNA  | 21,594,265 | 1,317,250,165 | 61      | 61      | 61      |
| K27ac_9-12-3_liver_R1.fastq.gz | FASTQ  | DNA  | 36,874,920 | 2,249,370,120 | 61      | 61      | 61      |
| K27ac_9-12-3_liver_R2.fastq.gz | FASTQ  | DNA  | 36,874,920 | 2,249,370,120 | 61      | 61      | 61      |
| K27ac_9-2-1_liver_R1.fastq.gz  | FASTQ  | DNA  | 42,629,512 | 2,600,400,232 | 61      | 61      | 61      |
| K27ac_9-2-1_liver_R2.fastq.gz  | FASTQ  | DNA  | 42,629,512 | 2,600,400,232 | 61      | 61      | 61      |
| K27ac_9-2-2_liver_R1.fastq.gz  | FASTQ  | DNA  | 44,702,678 | 2,726,863,358 | 61      | 61      | 61      |
| K27ac_9-2-2_liver_R2.fastq.gz  | FASTQ  | DNA  | 44,702,678 | 2,726,863,358 | 61      | 61      | 61      |
| K27ac_9-2-3_liver_R1.fastq.gz  | FASTQ  | DNA  | 42,647,989 | 2,601,527,329 | 61      | 61      | 61      |
| K27ac_9-2-3_liver_R2.fastq.gz  | FASTQ  | DNA  | 42,647,989 | 2,601,527,329 | 61      | 61      | 61      |
| K27ac_9-6-3_liver_R1.fastq.gz  | FASTQ  | DNA  | 26,256,657 | 1,601,656,077 | 61      | 61      | 61      |
| K27ac_9-6-3_liver_R2.fastq.gz  | FASTQ  | DNA  | 26,256,657 | 1,601,656,077 | 61      | 61      | 61      |

